test/e2e/components/Application.test.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+    describe,
+    expect,
+    it,
+    vi,
+} from 'vitest'
+import { type Application as PixiApplication } from 'pixi.js';
+import { render } from '@testing-library/react';
+import { useEffect } from 'react';
+
+import { Application } from '../../../src/components/Application';
+import { isAppMounted } from '../../utils/isAppMounted';
+import { useApplication } from '../../../src/hooks/useApplication';
+
+describe('Application', () => {
+    describe('onInit', () => {
```

Comment:
A test to check the size of `roots` and ensure that it decreases in size as expected after `unmountRoot` would probably be worthwhile

Reply:
I just wrote the test and it verifies the issue. I'm working on the solution now!